lib/server/error/wrapInErrorHandlerFactory.js

```diff
@@ -0,0 +1,57 @@
+const GrpcError = require('./GrpcError');
+const InternalGrpcError = require('./InternalGrpcError');
+
+/**
+ * @param {Error} error
+ * @param {grpc.ServerWriteableStream} call
+ * @param {function(Error, *)} [callback]
+ */
+function respondWithError(error, call, callback = undefined) {
+  if (callback) {
+    callback(error, null);
+  } else {
+    call.destroy(error);
+  }
+}
+
+/**
+ * @param {Object} logger
+ * @return wrapInErrorHandler
+ */
+module.exports = function wrapInErrorHandlerFactory(logger) {
+  /**
+   * Wrap RPC method in error handler
+   *
+   * @typedef wrapInErrorHandler
+   * @param {Function} method RPC method
+   * @return {Function}
+   */
+  function wrapInErrorHandler(method) {
+    /**
+     * @param {grpc.ServerWriteableStream} call
+     * @param {function(Error, *)} [callback]
+     */
+    async function rpcMethodErrorHandler(call, callback = undefined) {
+      try {
+        const result = await method(call);
+
+        if (callback) {
+          callback(null, result);
+        }
+      } catch (e) {
+        if (e instanceof GrpcError) {
+          respondWithError(e, call, callback);
```

Comment:
In case if `e` is `InternalGrpcError` we won't log it. You can do something like that https://github.com/dashevo/js-machine-abci/pull/1/files#diff-9f0554bbb0dfb0478de14bd5b365b699